src/OrchardCore/OrchardCore.Abstractions/Shell/Builders/ShellContext.cs

```diff
@@ -85,13 +88,18 @@ public ShellScope CreateScope()
         /// <summary>
         /// Mark the <see cref="ShellContext"/> as released and then a candidate to be disposed.
         /// </summary>
-        public void Release() => ReleaseInternal();
+        public void Release() => ReleaseInternalAsync().GetAwaiter().GetResult();
```

Comment:
We should either duplicate the previous implementation or mark it `[Obsolete]`. We don't want existing code to call this `GetAwaiter().GetResult()`

Reply:
Quite long to duplicate and will have to call `TerminateShellAsync().GetAwaiter().GetResult()` as before, so I would prefer to make it obsolete, but still used in `AddDependentShell()`, I thought about adding `AddDependentShellAsync()` but only used to config OpenId options, so synchronously.

We also have a `CreateScope()` that calls `TerminateShellAsync().GetAwaiter().GetResult()` if the shell is already released, we may also need a `CreateScopeAsync()`.

I will look at it.

Reply:
Finally, even if it is not so useful, I also made AddDependentShell() async ;)

Reply:
i was hesitating but I also changed `ShellContext.CreateScope()` to `.CreateScopeAsync()`.